src/base/string_utils.cpp

```diff
@@ -20,4 +20,9 @@ std::vector<std::string> split(std::string const &input, char delimiter) {
   return output;
 }
 
+bool startsWith(std::string_view input, std::string_view prefix)
+{
+  return input.size() >= prefix.size() && input.substr(0, prefix.size()) == prefix;
```

Comment:
This is simple and easy to read 👍🏻 so I would say it's fine if it's a little slower compared to the Boost version. We don't use it that frequently. And once we move to C++ 20, we can switch to the one from the standard lib (https://en.cppreference.com/w/cpp/string/basic_string/starts_with).

Reply:
You gave me an awesome idea! Since C++20 is already partially supported by gcc/clang, I looked at how an implementation of this looks like and I made a small change that ended up being about 30% faster than the previous one, whilst remaining  readable (and extremely fast in debug mode - almost the same as in release actually).

Reply:
Nice one! 👍🏻

Reply:
Just out of curiosity, I had a quick look (https://github.com/boostorg/algorithm/blob/c9077bd49580bddfaa8a5c71abc4022a2b09d972/include/boost/algorithm/string/predicate.hpp) but I'm not really sure what makes it faster 🤔 it's more or less the same, except that they check both iterators against their respective range end on each iteration, instead of comparing the sizes upfront like you do. I would expect the latter to be faster since there should be fewer instructions per loop iteration, but who knows.. Could be interesting to compare the generated assembly.

In any case, your new version looks great to me 👍🏻